src/git.md

```diff
@@ -0,0 +1,181 @@
+# Using git
+
+The Rust project uses [git][git] to manage its source code. In order to
+contribute, you'll need some familiarity with its features so that your changes
+can be incorporated into the compiler.
+
+[git]: https://git-scm.com
+
+The goal of this page is to cover some of the more common questions and
+problems new contributors face. Although some git basics will be covered here,
+if you have never used git or GitHub before you may find that this is still a
+little too fast for you. In that case, it would make sense to first read some
+introductions to git, such as the Beginner and Getting started sections of
+[this tutorial from Atlassian][atlassian-git]. GitHub also provides
+documentation and guides for beginners.
+
+[atlassian-git]: https://www.atlassian.com/git/tutorials/what-is-version-control
+
+Although this page should get you to a point where you can start contributing,
+learning more git is almost certainly a good use of your time if you want to
+keep contributing. There are many tutorials online for those folks that are
+newer which combine excellently with man pages and official documentation.
+
+## Prequisites
+
+We'll assume that you've installed git, forked rust-lang/rust, and cloned the
+forked repo to your PC. We'll also use the command line interface to interact
+with git; there are also a number of GUIs and IDE integrations that can
+generally do the same things.
+
+If you've cloned your fork, then you will be able to reference it with `origin`
+in your local repo. It may be helpful to also set up a remote for the official
+rust-lang/rust repo via
+
+```sh
+git remote add rust git@github.com:rust-lang/rust.git
+```
+
+if you're using SSH, or
+
+```sh
+git remote add rust https://github.com/rust-lang/rust.git
+```
+
+if you're using HTTPS.
+
+## Standard Process
+
+Below is the normal procedure that you're likely to use for most minor changes
+and PRs:
+
+ 1. Ensure that you're making your changes on top of master:
+ `git checkout master`.
+ 2. Get the latest changes from the Rust repo: `git pull rust master`.
+ 3. Make a new branch for your change: `git checkout -b issue-12345-fix`.
+ 4. Make some changes to the repo and test them.
+ 5. Stage your changes via `git add src/changed/file.rs src/another/change.rs`
+ and then commit them with `git commit`. Of course, making intermediate commits
+ may be a good idea as well. Avoid `git add .`, as it makes it too easy to
+ unintentionally commit changes that should not be committed. You can use
```

Comment:
We've had so much trouble with submodules lol, I think there's some suggestions drifting around to switch to `subtree` it's so bad.
```suggestion
 unintentionally commit changes that should not be committed, such as submodule updates. You can use
```